src/checkbox/Checkbox.ts

```diff
@@ -89,31 +107,35 @@ export default class Checkbox extends CheckboxBase<CheckboxProperties> {
 			required ? css.required : null
 		];
 
-		const checkbox = v('div', { classes: this.classes(css.inputWrapper) }, [
-			v('input', {
-				bind: this,
-				classes: this.classes(css.input),
-				checked,
-				'aria-describedby': describedBy,
-				disabled,
-				'aria-invalid': invalid + '',
-				name,
-				readOnly,
-				'aria-readonly': readOnly ? 'true' : null,
-				required,
-				type: 'checkbox',
-				value,
-				onblur: this._onBlur,
-				onchange: this._onChange,
-				onclick: this._onClick,
-				onfocus: this._onFocus,
-				onmousedown: this._onMouseDown,
-				onmouseup: this._onMouseUp,
-				ontouchstart: this._onTouchStart,
-				ontouchend: this._onTouchEnd,
-				ontouchcancel: this._onTouchCancel
-			})
-		]);
+		const children = [
+			mode === Mode.toggle && offLabel ? v('div', { classes: this.classes(css.offLabel) }, [ offLabel ]) : null,
+			v('div', { classes: this.classes(css.inputWrapper) }, [
+				v('input', {
+					bind: this,
+					classes: this.classes(css.input),
+					checked,
+					'aria-describedby': describedBy,
+					disabled,
+					'aria-invalid': invalid + '',
+					name,
+					readOnly,
+					'aria-readonly': readOnly ? 'true' : null,
+					required,
+					type: 'checkbox',
+					value,
+					onblur: this._onBlur,
+					onchange: this._onChange,
+					onclick: this._onClick,
+					onfocus: this._onFocus,
+					onmousedown: this._onMouseDown,
+					onmouseup: this._onMouseUp,
+					ontouchstart: this._onTouchStart,
+					ontouchend: this._onTouchEnd,
+					ontouchcancel: this._onTouchCancel
+				})
+			]),
+			mode === Mode.toggle && onLabel ? v('div', { classes: this.classes(css.onLabel) }, [ onLabel ]) : null
```

Comment:
I'm also not sure they need to be on either side of the input. It's not really an issue, but some little OCD part of my brain would feel better if they were next to each other. Also, a quick search of google images for "toggle switch ui" shows "On" before "Off" in most cases.